File: Android/app/src/main/java/com/dreamproject/inwords/data/source/webService/session/TokenResponse.java
package com.dreamproject.inwords.data.source.webService.session;

import com.google.gson.annotations.SerializedName;

import androidx.annotation.NonNull;

public class TokenResponse {
    @NonNull
    @SerializedName("token")
    private String token;
    @SerializedName("userId")
    private int userId;

    TokenResponse() {
        this.token = "";
        this.userId = 0;
    }

    private TokenResponse(@NonNull String token, @NonNull int userId) {
        this.token = token;
        this.userId = userId;
    }

    public static TokenResponse errorToken() {
        return new TokenResponse("error_token", 0);
    }

    public boolean isValid() {
        return token.length() > 30; //TODO think about more convenient check
    }

    public String getBearer() {
        return "Bearer " + token;
    }

    public String getAccessToken() {
        return token;
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/data/source/remote/session/TokenResponse.kt
package ru.inwords.inwords.data.source.remote.session

import com.google.gson.annotations.SerializedName

class TokenResponse(
        @SerializedName("token") val accessToken: String = "",
        @SerializedName("userId") val userId: Long = 0
) {
    val bearer: String get() = "Bearer $accessToken"

    //TODO think about more convenient check
    val isValid: Boolean get() = accessToken.length > 30
}
